test/lib/Benchmark.js

```diff
@@ -0,0 +1,182 @@
+/* globals XPCOMUtils, Services, Task */
+"use strict";
+
+const {Cu} = require("chrome");
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.import("resource://gre/modules/Services.jsm");
+
+XPCOMUtils.defineLazyModuleGetter(this, "Task",
+                                  "resource://gre/modules/Task.jsm");
+// For the performance.now()
+const performance = Services.appShell.hiddenDOMWindow.performance;
+
+/*
+ * A simple benchmark framework that runs on top of jpm test
```

Comment:
Out of curiousity what are the specific use cases here, besides measuring (eg) tab-open times?